### PR TITLE
bake sprintf into debugmes

### PR DESCRIPTION
**Pull request by mekolat**
I frequently find myself doing things like `debugmes(sprintf("foo=%d, bar=%d", .@foo, .@bar));` so I thought "why not simplify this?".

```c
debugmes("Hello %s!", "World"); // => Hello World!
```

### CODE REVIEW

**Comment by HerculesWSAPI** [comment]
This change is [Reviewable](https://reviewable.io/reviews/herculesws/hercules/2146)

File: src/map/script.c [hunk 1]
```diff
@@ -12349,9 +12349,18 @@ static BUILDIN(getstatus)
...
+
+	if (!script->sprintf_helper(st, 2, &buf)) {
+		StrBuf->Destroy(&buf);
+		script_pushint(st, 0);
+		return false;
+	}
+
+	ShowDebug("script debug : %d %d : %s\n", st->rid, st->oid, StrBuf->Value(&buf));
+	StrBuf->Destroy(&buf);
+	script_pushint(st, 1);
```
Comment (AnnieRuru) [inline src/map/script.c]: whats the `script_pushint` doing here ?
its not mention anywhere in the documentation

I guess its to return value based on getting use of sprintf_helper ?

Comment (mekolat) [reply src/map/script.c]: it's to return true/false whether or not the message is printed to the console, because `sprintf_helper` can fail

Comment (mekolat) [reply src/map/script.c]: there you go @AnnieRuru, I updated the docs

**Pull request merged**